great_expectations/data_context/data_context.py

```diff
@@ -1857,11 +1860,29 @@ def get_validator(
                 )
             )
 
-        return self.get_validator_using_batch_list(
+        expectation_suite.update_last_citation(
+            comment="Updated suite via context.get_validator",
+            batch_request_list=batch_request_list,
+        )
+
+        validator: Validator = self.get_validator_using_batch_list(
             expectation_suite=expectation_suite,
             batch_list=batch_list,
         )
 
+        # changes that validator made to the batch_request datasource config (e.g. add data_asset_name)
+        # are updated in project config and saved to disk
+        for datasource in validator.data_context.datasources:
+            self._instantiate_datasource_from_config_and_update_project_config(
+                name=datasource,
+                config=validator.data_context.datasources[datasource].config,
+                initialize=False,
+            )
+            if isinstance(self, DataContext):
+                self._save_project_config()
```

Comment:
Can we do a single save at the end of the loop? Why is this necessary with each iteration?

Reply:
Nice catch! I added the loop later and should have just indented the top part. ✅